packages/scheduler/lib/scheduler.ts

```diff
@@ -286,7 +289,8 @@ export class Scheduler {
      */
     public async setScheduleState({ scheduleName, state }: { scheduleName: string; state: ScheduleState }): Promise<Result<Schedule>> {
         return this.dbClient.db.transaction(async (trx) => {
-            const schedule = await schedules.search(trx, { name: scheduleName, limit: 1 });
+            // forUpdate = true so that the schedule is locked to prevent any concurrent update or concurrent scheduling of tasks
+            const schedule = await schedules.search(trx, { name: scheduleName, limit: 1, forUpdate: true });
```

Comment:
not using `skipLocked` here and above because it would fail. Therefore we just wait for the schedule to be unlocked. The drawback is that it might makes those calls slower